python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -0,0 +1,493 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy as np
+
+from lsstDebug import getDebugFrame
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.afw.math import BackgroundList
+from lsst.afw.table import SourceTable, SourceCatalog
+from lsst.meas.algorithms import estimateBackground
+from lsst.meas.algorithms.installGaussianPsf import InstallGaussianPsfTask
+from lsst.meas.astrom import AstrometryTask, displayAstrometry
+from .detectAndMeasure import DetectAndMeasureTask
+from .measurePsf import MeasurePsfTask
+from .repair import RepairTask
+
+class CharacterizeImageConfig(pexConfig.Config):
+    """!Config for CharacterizeImageTask"""
+    doMeasurePsf = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Measure PSF? If False then keep the existing PSF model (which must exist) "
+            "and use that model for all operations."
+    )
+    doWrite = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Persist results?",
+    )
+    doWriteExposure = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Write icExp and icExpBackground in addition to icSrc? Ignored if doWrite False.",
+    )
+    psfIterations = pexConfig.RangeField(
+        dtype = int,
+        default = 2,
+        min = 1,
+        doc = "Number of iterations of detect sources, measure sources, estimate PSF. "
+            "If useSimplePsf='all_iter' then 2 should be plenty; otherwise more may be wanted.",
+    )
+    background = pexConfig.ConfigField(
+        dtype = estimateBackground.ConfigClass,
+        doc = "Configuration for initial background estimation",
+    )
+    detectAndMeasure = pexConfig.ConfigurableField(
+        target = DetectAndMeasureTask,
+        doc = "Detect and measure sources, for measuring PSF",
+    )
+    useSimplePsf = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Replace the existing PSF model with a simplified version that has the same sigma "
+            "at the start of each PSF determination iteration? Doing so makes PSF determination "
+            "converge more robustly and quickly.",
+    )
+    installSimplePsf = pexConfig.ConfigurableField(
+        target = InstallGaussianPsfTask,
+        doc = "Install a simple PSF model",
+    )
+    astrometry = pexConfig.ConfigurableField(
+        target = AstrometryTask,
+        doc = "Task to load and match reference objects. Only used if measurePsf can use matches. "
+            "Warning: matching will only work well if the initial WCS is accurate enough "
+            "to give good matches (roughly: good to 3 arcsec across the CCD).",
```

Comment:
This is far too specific.  Surely that warning is appropriate for any and all astrometric matching?
And how can you possibly say it needs to be good to 3 arcsec?  Surely it depends on the matching radius in the matcher?